File: frontend/src/app/models/call.request.ts
import { CallStatus, CallType } from "./call.enum";

export interface CallRequest {
  callId: string;
  callerId: string | null; // UUID as string
  recipientId?: string; // UUID as string
  conversationId: string; // UUID as string
  status: CallStatus;
  callType: CallType;
}

File: backend/src/main/java/com/btalk/dto/request/CallRequest.java
package com.btalk.dto.request;

import com.btalk.constants.CallStatus;
import com.btalk.constants.CallType;
import com.btalk.utils.UuidDeserializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import lombok.Data;

@Data
public class CallRequest {
    @JsonDeserialize(using = UuidDeserializer.class)
    private String callId;
    
    @JsonDeserialize(using = UuidDeserializer.class)
    private String callerId;
    
    @JsonDeserialize(using = UuidDeserializer.class)
    private String recipientId;
    
    @JsonDeserialize(using = UuidDeserializer.class)
    private String conversationId;
    
    private CallStatus status;
    private CallType callType; 
}